### ISSUE TITLE
Wall and UT indicators are not parsed correctly.

### ISSUE THREAD

**Issue by mniip** [ygale/timezone-olson#7]
I have a TZif v2 file that looks like this:

```
00000000  54 5a 69 66 32 00 00 00  00 00 00 00 00 00 00 00  |TZif2...........|
00000010  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000020  00 00 00 01 00 00 00 02  00 00 00 08 92 e6 92 48  |...............H|
00000030  01 ff ff fc 38 00 00 00  00 00 00 00 04 4c 4d 54  |....8........LMT|
00000040  00 47 4d 54 00 54 5a 69  66 32 00 00 00 00 00 00  |.GMT.TZif2......|
00000050  00 00 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000060  00 00 00 00 00 00 00 00  01 00 00 00 02 00 00 00  |................|
00000070  08 ff ff ff ff 92 e6 92  48 01 ff ff fc 38 00 00  |........H....8..|
00000080  00 00 00 00 00 04 4c 4d  54 00 47 4d 54 00 0a 47  |......LMT.GMT..G|
00000090  4d 54 30 0a                                       |MT0.|
```
(`/usr/share/zoneinfo/Africa/Abidjan` from `tzdata-2019b-1` on Arch Linux)

It's being parsed into the following `OlsonData`:
```
OlsonData {olsonTransitions = [Transition {transTime = -1830383032, transIndex = 1},Transition {transTime = -1830383032, transIndex = 1}], olsonTypes = [], olsonLeaps = [], olsonPosixTZ = Just "GMT0"}
```
Note that `olsonTypes` is empty while `olsonTransitions` references into it, this is obviously incorrect.

The source of the issue seems to be that the v1 header contains, in RFC8536 jargon, `typecnt=2` but `isutcnt=0` and `isstdcnt=0`. The RFC states:

>   standard/wall indicators:  A series of one-octet values indicating
>      whether the transition times associated with local time types were
>      specified as standard time or wall-clock time.  Each value MUST be
>      0 or 1.  A value of one (1) indicates standard time.  The value
>      MUST be set to one (1) if the corresponding UT/local indicator is
>      set to one (1).  A value of zero (0) indicates wall time.  The
>      number of values is specified by the "isstdcnt" field in the
>      header.  **If "isstdcnt" is zero (0), all transition times
>      associated with local time types are assumed to be specified as
>      wall time.**
>
>   UT/local indicators:  A series of one-octet values indicating whether
>      the transition times associated with local time types were
>      specified as UT or local time.  Each value MUST be 0 or 1.  A
>      value of one (1) indicates UT, and the corresponding standard/wall
>      indicator MUST also be set to one (1).  A value of zero (0)
>      indicates local time.  The number of values is specified by the
>      "isutcnt" field in the header.  **If "isutcnt" is zero (0), all
>      transition times associated with local time types are assumed to
>      be specified as local time.**

Emphasized part is suggesting that the file above is valid, and should be interpreted as if all transitions happened in Wall time.

The related logic is implemented in https://github.com/ygale/timezone-olson/blob/master/Data/Time/LocalTime/TimeZone/Olson/Parse.hs#L152-L165 where `zipWithExtend` is used in case either the `isstds` or the `isgmts` is shorter than the other. However then regular `zipWith` is used to join the resulting list with the list of `ttinfos`.
 
Thus the case when both `isstds` and `isgmts` are shorter than `ttinfos` is not handled and we get a truncated list of types as above.

**Comment by ygale** [comment]
Thanks for this report. RFC8536 is new, and this appears to be a change to the original TZ info file format specification, at least the way I interpreted it originally. As you can see, I intentionally went to some trouble to discard any ttinfos beyond the isstds and isgmts, treating them as malformed corrupted data.

If the spec now supports this and such files are not corrupted, then your change makes perfect sense.

**Comment by mniip** [comment]
By the way the CI failure seems to be irrelevant to the changes

**Comment by mniip** [comment]
@ygale any update? This is causing issues for us.

**Comment by ygale** [comment]
Sorry, I was the internal support "duty officer" at work for the sprint that ends today. Hope to get to this soon.

**Comment by mniip** [comment]
@ygale Sorry to disturb but is there any update?

**Comment by mniip** [comment]
Looks like this is fixed in 4ff152bd4f5e1949ec31834b87a5b19aff8a04b0. Can we get a hackage release please?

**Comment by ygale** [comment]
Thanks, released as 0.2.0 on hackage. Sorry for the delay.
